app/models/aliquot.rb

```diff
@@ -59,4 +59,43 @@ def tagged?
   def collection?
     false
   end
+
+  # Returns the id to publish
+  def publish_id
+    id.to_s
+  end
+
+  # Returns the type of the source to publish
```

Comment:
Same here and below. Also why are using respond_to here? Is it necessary. It feels better to have a publish or publisher object. Also it might be better to do the to_s or '' in the source.

Reply:
I think this requires changes in DataStructure builder, similar to one I mentioned above. That's is the reason why I have made it work in such a way to avoid changes in Datastructure builder as this story has already gone beyond the size. But with the new suggestions, I think that is required any way, so let me see how these can be resolved.